File: src/main/java/io/github/saneea/dvh/feature/clipboard/Clipboard.java
package io.github.saneea.dvh.feature.clipboard;

import java.util.Map;
import java.util.function.Supplier;

import io.github.saneea.dvh.Feature;
import io.github.saneea.dvh.FeatureContext;
import io.github.saneea.dvh.feature.multi.MultiFeatureBase;

public class Clipboard extends MultiFeatureBase {

	@Override
	public Meta meta(FeatureContext context) {
		return Meta.from("read/write clipboard (text only)");
	}

	@Override
	public Map<String, Supplier<Feature>> getFeatureAliases() {
		return new AliasesBuilder()//
				.feature("read", TextFromClipboard::new)//
				.feature("write", TextToClipboard::new)//
				.build();
	}

}

File: src/main/java/io/github/saneea/dvh/feature/clipboard/Clipboard.kt
package io.github.saneea.dvh.feature.clipboard

import io.github.saneea.dvh.Feature.Meta
import io.github.saneea.dvh.FeatureContext
import io.github.saneea.dvh.feature.multi.FeaturesCreators
import io.github.saneea.dvh.feature.multi.MultiFeatureBase

class Clipboard : MultiFeatureBase() {

    override fun meta(context: FeatureContext) = Meta("read/write clipboard (text only)")

    override fun getFeatureAliases(): FeaturesCreators {
        return AliasesBuilder()
            .feature("read", ::TextFromClipboard)
            .feature("write", ::TextToClipboard)
            .build()
    }
}